playback/service/src/main/java/de/danoeh/antennapod/playback/service/internal/PlaybackServiceTaskManager.java

```diff
@@ -148,22 +143,17 @@ public synchronized void setSleepTimer(long waitingTime) {
 
         Log.d(TAG, "Setting sleep timer to " + waitingTime + " milliseconds");
         if (isSleepTimerActive()) {
-            sleepTimerFuture.cancel(true);
+            sleepTimer.reset(waitingTime);
+        } else {
+            sleepTimer = SleepTimerFactory.createSleepTimer(context, waitingTime);
         }
-        sleepTimer = new SleepTimer(waitingTime);
-        sleepTimerFuture = schedExecutor.schedule(sleepTimer, 0, TimeUnit.MILLISECONDS);
-        EventBus.getDefault().post(SleepTimerUpdatedEvent.justEnabled(waitingTime));
     }
 
     /**
      * Returns true if the sleep timer is currently active.
      */
     public synchronized boolean isSleepTimerActive() {
-        return sleepTimer != null
-                && sleepTimerFuture != null
-                && !sleepTimerFuture.isCancelled()
-                && !sleepTimerFuture.isDone()
-                && sleepTimer.getWaitingTime() > 0;
+        return sleepTimer != null && sleepTimer.isActive();
     }
```

Comment:
My long-term goal is to get rid of the TaskManager completely. Instead, please move these methods to be a method on the sleep timer itself, and add a sleep timer attribute to the PlaybackService